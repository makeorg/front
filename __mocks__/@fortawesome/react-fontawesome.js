
import React from 'react';

export const FontAwesomeIcon = ({ icon }) => (
  <svg
    lib="@fortawesome/react-fontawesome"
    icon={icon.iconName}
    prefix={icon.prefix}
  />
);
