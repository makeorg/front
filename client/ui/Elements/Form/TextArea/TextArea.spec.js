import React from 'react';
import snapshotDiff from 'snapshot-diff';
import renderer from 'react-test-renderer';
import { faPenAlt } from '@fortawesome/free-solid-svg-icons';
import { TextArea } from './index';

jest.mock('../Styled/Input', () => ({
  FakeInputStyle: 'FakeInputStyle',
}));

jest.mock('../Styled/TextArea', () => ({
  TextAreaIconStyle: 'TextAreaIconStyle',
  BasicTextAreaStyle: 'BasicTextAreaStyle',
}));

const defaultProps = {
  name: 'bar',
  icon: faPenAlt,
  value: 'baz',
  label: 'qux',
  handleChange: () => {},
};

describe('TextArea', () => {
  it('must return the diff between snapshot with a negative vs a postive Tab Index', () => {
    const NegativeTabIndex = renderer
      .create(<TextArea {...defaultProps} tabIndex="-1" />)
      .toJSON();
    const PositiveTabIndex = renderer
      .create(<TextArea {...defaultProps} tabIndex="0" />)
      .toJSON();
    expect(snapshotDiff(NegativeTabIndex, PositiveTabIndex)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea is required or optionnal', () => {
    const RequiredTextArea = renderer
      .create(<TextArea {...defaultProps} required />)
      .toJSON();
    const OptionnalTextArea = renderer
      .create(<TextArea {...defaultProps} required={false} />)
      .toJSON();
    expect(snapshotDiff(RequiredTextArea, OptionnalTextArea)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a limit of characters or not', () => {
    const NoLimitTextArea = renderer
      .create(<TextArea {...defaultProps} />)
      .toJSON();
    const LimitedTextArea = renderer
      .create(<TextArea {...defaultProps} minLength="25" maxLength="450" />)
      .toJSON();
    expect(snapshotDiff(NoLimitTextArea, LimitedTextArea)).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a enable vs disable spellchecking', () => {
    const EnabledSpellchecking = renderer
      .create(<TextArea {...defaultProps} />)
      .toJSON();
    const DisabledSpellchecking = renderer
      .create(<TextArea {...defaultProps} spellCheck={false} />)
      .toJSON();
    expect(
      snapshotDiff(EnabledSpellchecking, DisabledSpellchecking)
    ).toMatchSnapshot();
  });

  it('must return the diff between snapshot when TextArea has a disable vs enable autocomplete', () => {
    const DisabledAutocomplete = renderer
      .create(<TextArea {...defaultProps} />)
      .toJSON();
    const EnabledAutocomplete = renderer
      .create(<TextArea {...defaultProps} autoComplete="on" />)
      .toJSON();
    expect(
      snapshotDiff(DisabledAutocomplete, EnabledAutocomplete)
    ).toMatchSnapshot();
  });
});
