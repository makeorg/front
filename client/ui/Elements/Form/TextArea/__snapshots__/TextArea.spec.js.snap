// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea must return the diff between snapshot when TextArea has a disable vs enable autocomplete 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -10,11 +10,11 @@
        SvgPencil
      </span>
    </label>
    <BasicTextAreaStyle
      aria-required={false}
-     autoComplete="off"
+     autoComplete="on"
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
`;

exports[`TextArea must return the diff between snapshot when TextArea has a enable vs disable spellchecking 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -17,10 +17,10 @@
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
      rows={5}
-     spellCheck={true}
+     spellCheck={false}
      tabIndex={0}
      value="baz"
    />
  </FakeInputStyle>
`;

exports[`TextArea must return the diff between snapshot when TextArea has a limit of characters or not 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -12,10 +12,12 @@
    </label>
    <BasicTextAreaStyle
      aria-required={false}
      autoComplete="off"
      id="bar"
+     maxLength="450"
+     minLength="25"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
      rows={5}
`;

exports[`TextArea must return the diff between snapshot when TextArea is required or optionnal 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -9,17 +9,17 @@
      >
        SvgPencil
      </span>
    </label>
    <BasicTextAreaStyle
-     aria-required={true}
+     aria-required={false}
      autoComplete="off"
      id="bar"
      name="bar"
      onChange={[Function handleChange]}
      placeholder="qux"
-     required={true}
+     required={false}
      rows={5}
      spellCheck={true}
      tabIndex={0}
      value="baz"
    />
`;

exports[`TextArea must return the diff between snapshot with a negative vs a postive Tab Index 1`] = `
Snapshot Diff:
- First value
+ Second value

@@ -18,9 +18,9 @@
      onChange={[Function handleChange]}
      placeholder="qux"
      required={false}
      rows={5}
      spellCheck={true}
-     tabIndex="-1"
+     tabIndex="0"
      value="baz"
    />
  </FakeInputStyle>
`;
